src/frontend/src/pages/GanttPage/Gantt.tsx

```diff
@@ -0,0 +1,229 @@
+import { addDays, differenceInDays, eachDayOfInterval, isMonday } from 'date-fns';
+import { ComponentProps, DragEvent, MouseEvent, useEffect, useState } from 'react';
+import useMeasure from 'react-use-measure';
+import {
+  GANTT_CHART_GAP_SIZE,
+  GANTT_CHART_CELL_SIZE,
+  applyChangesToEvents,
+  EventChange,
+  Task
+} from '../../utils/gantt.utils';
+import useId from '@mui/material/utils/useId';
+import { Box, Typography, useTheme } from '@mui/material';
+import { grey } from '@mui/material/colors';
+import { dateToString } from '../../utils/datetime.utils';
+
+interface GanttProps {
+  start: Date;
+  end: Date;
+  tasks: Task[];
+  isEditMode: boolean;
+  saveChanges: (eventChanges: EventChange[]) => void;
+}
+
+export function Gantt({ start, end, tasks, isEditMode, saveChanges }: GanttProps) {
```

Comment:
see comment below, but if this represents a team in the gantt chart, it should be named GanttChartSection or GanttChartTeamSection, be an arrow function component and be in it's own file